user/contactmethod/store.go

```diff
@@ -105,6 +127,65 @@ func NewDB(ctx context.Context, db *sql.DB) (*DB, error) {
 	}, p.Err
 }
 
+func (db *DB) MetadataByTypeValue(ctx context.Context, tx *sql.Tx, typ Type, value string) (*Metadata, error) {
+	err := permission.LimitCheckAny(ctx, permission.Admin)
+	if err != nil {
+		return nil, err
+	}
+	var data json.RawMessage
+	var t time.Time
+	err = wrapTx(ctx, tx, db.metaTV).QueryRowContext(ctx, typ, value).Scan(&data, &t)
+	if err != nil {
+		return nil, err
+	}
+
+	var m Metadata
+	err = json.Unmarshal(data, &m)
+	if err != nil {
+		return nil, err
+	}
+	m.FetchedAt = t
+
+	return &m, nil
+}
+
+func (db *DB) SetCarrierV1MetadataByTypeValue(ctx context.Context, tx *sql.Tx, typ Type, value string, newM *Metadata) error {
+	err := permission.LimitCheckAny(ctx, permission.Admin)
+	if err != nil {
+		return err
+	}
+	var ownTx bool
+	if tx == nil {
+		tx, err = db.db.BeginTx(ctx, nil)
+		if err != nil {
+			return err
+		}
+		defer tx.Rollback()
+		ownTx = true
+	}
+	m, err := db.MetadataByTypeValue(ctx, tx, typ, value)
+	if err != nil {
+		return err
+	}
+	m.CarrierV1 = newM.CarrierV1
+	m.CarrierV1.UpdatedAt = m.FetchedAt
+
+	data, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	_, err = tx.StmtContext(ctx, db.setMetaTV).ExecContext(ctx, typ, value, data)
+	if err != nil {
+		return err
+	}
+
+	if ownTx {
+		return tx.Commit()
+	}
```

Comment:
can you `defer tx.Commit()` inside of the if statement above instead of needing to use a bool flag?

Reply:
A `Commit` call should only be called once and needs to be checked for errors (as opposed to rollback). In this case we don't want to call commit if we were passed a `tx` because the caller would expect to keep using the transaction.

We only want to commit if we started/own the transaction inside this method. Otherwise it's up to the caller to commit/rollback their own transaction.